rgdps/api/gd/user_relationships.py

```diff
@@ -0,0 +1,282 @@
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to accept friend request from {target_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully accepted friend request from {target_id}.")
+    return responses.success()
+
+
+async def user_relatoionship_list_get(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    relationship_type: UserRelationshipType = Form(..., alias="type"),
+):
+
+    result = await user_relationships.get_user(
+        ctx,
+        user.id,
+        relationship_type,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to view user relationship of {user.id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    if not result.relationships:  # No relationships found.
+        return responses.code(-2)
+
+    response = "|".join(
+        gd_obj.dumps(
+            [
+                gd_obj.create_profile(relationship.user),
+                gd_obj.create_user_relationship_list(relationship.relationship),
+            ],
+        )
+        for relationship in result.relationships
+    )
+
+    # Mark them as seen.
+    await user_relationships.mark_all_as_seen(ctx, user.id, relationship_type)
+    logger.info(f"{user} successfully viewed user relationship.")
+
+    return response
+
+
+async def user_friend_remove(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+):
+    result = await user_relationships.remove_friendship(
+        ctx,
+        user.id,
+        target_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to remove friend {target_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully removed friend {target_id}.")
+    return responses.success()
+
+
+async def block_user(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+):
+
+    # Remove friendship if exists.
+    friendship = await user_relationships.get_user(
+        ctx,
+        user.id,
+        UserRelationshipType.FRIEND,
+    )
+
+    if not isinstance(friendship, ServiceError):
+        await user_relationships.remove_friendship(
+            ctx,
+            user.id,
+            target_id,
+        )
+
+    result = await user_relationships.create(
+        ctx,
+        user.id,
+        target_id,
+        UserRelationshipType.BLOCKED,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to block user {target_id} with error {result!r}.",
+        )
+    else:
+        logger.info(f"{user} successfully blocked user {target_id}.")
+
+    return responses.success()
+
+
+async def unblock_user(
```

Comment:
Would call `unblock_user_post` for consistency's sake

Reply:
Resolved in 5ef80ff